webpages/js/metaanalyses.js

```diff
@@ -512,15 +516,58 @@
           delete line.ucl;
           delete line.wt;
         }
+
+        lines.push(line);
       }
     }
 
     // add indication to the graph when there is no data
     if (lines.length === 0) {
-      var noDataLine = {title: "No data"};
+      var noDataLine = {
+        title: "No data",
+        group: "No data"
+      };
+      groups.push(noDataLine.group)
       lines.push(noDataLine);
     }
 
+    groups.sort(); // alphabetically
+
+    // calculating wtg
+    groups.forEach(function (group) {
+      var sumOfWtg = 0;
+      lines.forEach(function (line) {
+        if (line.group == group) {
+          sumOfWtg += line.wt;
+        }
+      });
+
+      lines.forEach(function (line) {
+        if (line.group == group) {
+          line.wtg = line.wt / sumOfWtg * 1000;
+        }
+      });
+    });
+
+    var dataGroups =
+      groups.map(function (group) {
+        return lines.filter(function (exp) { return exp.group == group; });
+      });
+
+    // todo use per-group aggregates here
+    var perGroup = {};
+    dataGroups.forEach(function (dataGroup) {
+      perGroup[dataGroup[0].group] = {};
+      perGroup[dataGroup[0].group].wt = dataGroup.reduce(function sum(acc, line) {return line.wt != null ? acc+line.wt : acc;}, 0);
+      if (perGroup[dataGroup[0].group].wt == 0) perGroup[dataGroup[0].group].wt = 1;
+      perGroup[dataGroup[0].group].or = dataGroup.reduce(function sumproduct(acc, line) {return line.wt != null ? acc+line.or*line.wt : acc;}, 0) / perGroup[dataGroup[0].group].wt;
+    });
+
+    // todo highlight the current experiment in forest plot and in grape chart
+    // set chart width based on number of groups
```

Comment:
we don't need to do anything with chart width, drop this bit; height (and viewBox for that) should be handled somewhere else anyway